tests/integration/awslambda/test_lambda_integration.py

```diff
@@ -420,6 +420,139 @@ def verify_failure_received():
         finally:
             lambda_client.delete_event_source_mapping(UUID=event_source_mapping_uuid)
 
+    @pytest.mark.parametrize(
+        "item_to_put1, item_to_put2, filter, calls",
+        [
+            # Test with no filter, and two times same entry (aka MODIFY)
+            (
+                {"id": {"S": "test123"}},
+                None,
+                None,
+                2,
+            ),
+            # Test with filter, and two times same entry (aka MODIFY)
+            (
+                {"id": {"S": "test123"}},
+                None,
+                {"eventName": ["INSERT"]},
+                1,
+            ),
+            # Test with OR filter, and two times same entry (aka MODIFY)
+            (
+                {"id": {"S": "test123"}},
+                None,
+                {"eventName": ["INSERT", "MODIFY"]},
+                2,
+            ),
+            # Test with 2 filters (AND), and two times same entry (aka MODIFY)
+            (
+                {"id": {"S": "test123"}},
+                None,
+                {"eventName": ["INSERT"], "eventSource": ["aws:dynamodb"]},
+                1,
+            ),
+            # Test exists filter
+            (
+                {"id": {"S": "test123"}},
+                {"id": {"S": "test1234"}, "presentKey": {"S": "test123"}},
+                {"dynamodb": {"NewImage": {"presentKey": [{"exists": False}]}}},
+                1,
+            ),
+            # numeric filters
+            (
+                {"id": {"S": "test123"}, "numericFilter": {"N": "123"}},
+                {"id": {"S": "test1234"}, "numericFilter": {"N": "12"}},
+                {"dynamodb": {"NewImage": {"numericFilter": [{"numeric": [">", 100]}]}}},
+                1,
+            ),
+            (
+                {"id": {"S": "test123"}, "numericFilter": {"N": "100"}},
+                {"id": {"S": "test1234"}, "numericFilter": {"N": "12"}},
+                {"dynamodb": {"NewImage": {"numericFilter": [{"numeric": [">=", 100, "<", 200]}]}}},
+                1,
+            ),
+            # Prefix
+            (
+                {"id": {"S": "test123"}, "prefix": {"S": "us-1-testtest"}},
+                {"id": {"S": "test1234"}, "prefix": {"S": "testtest"}},
+                {"dynamodb": {"NewImage": {"prefix": [{"prefix": "us-1"}]}}},
+                1,
+            ),
+        ],
+    )
+    def test_dynamo_db_insert_event(
+        self,
+        create_lambda_function,
+        lambda_client,
+        dynamodb_client,
+        dynamodb_create_table,
+        filter,
+        calls,
+        item_to_put1,
+        item_to_put2,
+    ):
+
```

Comment:
We need to create role and policy here. The `role_arn` will be used for the lambda in the next step:
```
        role = f"test-lambda-dynamodb-role-{short_uid()}"
        policy_name = f"test-lambda-dynamodb-role-policy-{short_uid()}"
        role_arn = create_iam_role_with_policy(
            RoleName=role,
            PolicyName=policy_name,
            RoleDefinition=lambda_role,
            PolicyDefinition=s3_lambda_permission,
        )
```